lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -1175,71 +1243,55 @@ def compute_severity(resource_type, record, entry_common_labels)
+        subfields[destination_key] = value
+      end
+
+      return unless subfields
 
-    def set_http_request(record, entry)
-      return nil unless record['httpRequest'].is_a?(Hash)
-      input = record['httpRequest']
       if @use_grpc
-        output = Google::Logging::Type::HttpRequest.new
+        output = Object.const_get(cfg[:grpc_class]).new
       else
-        output = Google::Apis::LoggingV2beta1::HttpRequest.new
+        output = Object.const_get(cfg[:nongrpc_class]).new
       end
-      # We need to delete each field from 'httpRequest' even if its value is
-      # nil. However we do not want to assign this nil value to the constructed
-      # json or proto.
-      request_method = input.delete('requestMethod')
-      output.request_method = request_method unless request_method.nil?
-      request_url = input.delete('requestUrl')
-      output.request_url = request_url unless request_url.nil?
-      request_size = input.delete('requestSize')
-      output.request_size = request_size.to_i unless request_size.nil?
-      status = input.delete('status')
-      output.status = status.to_i unless status.nil?
-      response_size = input.delete('responseSize')
-      output.response_size = response_size.to_i unless response_size.nil?
-      user_agent = input.delete('userAgent')
-      output.user_agent = user_agent unless user_agent.nil?
-      remote_ip = input.delete('remoteIp')
-      output.remote_ip = remote_ip unless remote_ip.nil?
-      referer = input.delete('referer')
-      output.referer = referer unless referer.nil?
-      cache_hit = input.delete('cacheHit')
-      output.cache_hit = cache_hit unless cache_hit.nil?
-      cache_validated_with_origin_server = \
-        input.delete('cacheValidatedWithOriginServer')
-      output.cache_validated_with_origin_server = \
-        cache_validated_with_origin_server \
-        unless cache_validated_with_origin_server.nil?
-
-      latency = input.delete('latency')
-      unless latency.nil?
-        # Parse latency. If no valid format is detected, skip setting latency.
-        # Format: whitespace (optional) + integer + point & decimal (optional)
-        #       + whitespace (optional) + "s" + whitespace (optional)
-        # e.g.: "1.42 s"
-        match = @compiled_http_latency_regexp.match(latency)
-        if match
-          # Split the integer and decimal parts in order to calculate seconds
-          # and nanos.
-          latency_seconds = match['seconds'].to_i
-          latency_nanos = (match['decimal'].to_f * NANOS_IN_A_SECOND).round
-          if @use_grpc
-            output.latency = Google::Protobuf::Duration.new(
-              seconds: latency_seconds,
-              nanos: latency_nanos
-            )
-          else
-            output.latency = {
-              seconds: latency_seconds,
-              nanos: latency_nanos
-            }.delete_if { |_, v| v == 0 }
-          end
+      subfields.each do |k, v|
```

Comment:
We normally try to avoid one-letter variable names if possible:

```
          subfields.each do |key, value|
            output.send("#{key}=", value)
          end
```